File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/Components/SDUI/SpoilYourselfTitleSDUI.swift
import SwiftUI

struct SpoilYourselfTitleSDUI: View {
    let component: SDUIComponent

    var body: some View {
        let title = component.prop(for: "title") as String? ?? "Spoil Yourself"
        let gradientHexColors =
            component.prop(for: "gradientColors") as [String]? ?? ["#FF6B00", "#FF6B00"]
        let gradientColors = gradientHexColors.map { Color(hex: $0) }

        // Adaptive background (if needed via style props)
        // For now, mirroring the reference implementation of just the title arrangement

        HStack {
            // Left Gradient Line
            LinearGradient(
                gradient: Gradient(colors: [.clear, gradientColors.first ?? .orange]),
                startPoint: .leading,
                endPoint: .trailing
            )
            .frame(height: 1)
            .frame(width: 60)

            // Sparkle
            Image(systemName: "sparkle")
                .font(.system(size: 12))
                .foregroundColor(gradientColors.first ?? .orange)

            // Title
            Text(title)
                .font(.system(size: 22, weight: .bold))
                .foregroundColor(.white)

            // Sparkle
            Image(systemName: "sparkle")
                .font(.system(size: 12))
                .foregroundColor(gradientColors.last ?? .orange)

            // Right Gradient Line
            LinearGradient(
                gradient: Gradient(colors: [gradientColors.last ?? .orange, .clear]),
                startPoint: .leading,
                endPoint: .trailing
            )
            .frame(height: 1)
            .frame(width: 60)
        }
        .padding(.horizontal, 16)
        .padding(.top, 0)  // Reduced from 8
        .padding(.bottom, 0)  // Reduced from 4
    }
}

File: apps/Applications/android/app/src/main/java/com/localforvocalstartup/app/ui/components/SpoilYourselfTitleSDUI.kt
package com.localforvocalstartup.app.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.localforvocalstartup.app.data.model.SDUIComponent

@Composable
fun SpoilYourselfTitleSDUI(component: SDUIComponent) {
    val props = component.props
    val title = props?.get("title")?.toString() ?: "Spoil Yourself"
    
    val gradientHexColors = props?.get("gradientColors") as? List<*> ?: listOf("#FF6B00", "#FF6B00")
    val gradientColors = gradientHexColors.map { hexRaw ->
        try {
            Color(android.graphics.Color.parseColor(hexRaw.toString()))
        } catch (e: Exception) {
            Color(0xFFFF6B00)
        }
    }

    val leadingColor = gradientColors.firstOrNull() ?: Color(0xFFFF6B00)
    val trailingColor = gradientColors.lastOrNull() ?: Color(0xFFFF6B00)

    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp),
        horizontalArrangement = Arrangement.Center,
        verticalAlignment = Alignment.CenterVertically
    ) {
        // Left Gradient Line
        Box(
            modifier = Modifier
                .width(60.dp)
                .height(1.dp)
                .background(
                    brush = Brush.horizontalGradient(
                        colors = listOf(Color.Transparent, leadingColor)
                    )
                )
        )

        Spacer(modifier = Modifier.width(8.dp))
        
        // Title
        Text(
            text = title,
            fontSize = 22.sp,
            fontWeight = FontWeight.Bold,
            color = Color.White
        )
        
        Spacer(modifier = Modifier.width(8.dp))

        // Right Gradient Line
        Box(
            modifier = Modifier
                .width(60.dp)
                .height(1.dp)
                .background(
                    brush = Brush.horizontalGradient(
                        colors = listOf(trailingColor, Color.Transparent)
                    )
                )
        )
    }
}
